src/bundle/DependencyInjection/Configuration.php

```diff
@@ -42,6 +43,17 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->prototype('scalar')->end()
                 ->end()
                 ->arrayNode('ip_whitelist')
+                    ->beforeNormalization()
+                        ->ifArray()
+                        ->then(static function (mixed $value): array {
+                            $values = [];
+                            foreach (self::flatten($value) as $v) {
+                                $values[] = $v;
+                            }
+
+                            return $values;
+                        })
```

Comment:
I believe this could be simplified with PHP on-board functions:

```
return iterator_to_array(new RecursiveIteratorIterator(new RecursiveArrayIterator($value)), false);
```

Reply:
Funny. I have actually used this in the past and completely forgot that it existed.